### DISCUSSION TITLE
Debugger: Go back to the previously active window before executing next line

### DISCUSSION THREAD

**Discussion by Gharwen** [mark-wiemer/ahkpp#424] [Q&A] [answered]
I try to debug a script with "send" instructions.
But the bad thing is, that VS Code will be the new active window when hitting one of the Step buttons and the "send" instruction will now hit in VS Code instead of the window for which the script is designed.
How can I achieve that the previously active window (for which the script is) will be active again before the next line is executed?

**Comment by fade2gray** [comment, marked as answer]
Do something like this ...

```ahk
MsgBox, Do stuff

MySendEvent() ; breakpoint here
MySendEvent(){
    WinActivate, Untitled - Notepad
    WinWaitActive, Untitled - Notepad
    SendEvent, foobar
}

MsgBox, Do more stuff
```

Set a breakpoint at the call to the `MySendEvent()` function and use `Step Over (F10)`.
